libraries/resource_providers/api200/volume_provider.rb

```diff
@@ -18,11 +18,12 @@ def load_resource_with_associated_resources
         if @new_resource.volume_template
           @item.set_storage_volume_template(resource_named(:VolumeTemplate).new(@item.client, name: @new_resource.volume_template))
         else # Can't set the storage_pool or snapshot_pool if we specify a volume_template
-          load_storage_system if @new_resource.storage_system
+          validate_required_properties(:storage_system, :storage_pool)
+          storage_system_uri = load_storage_system
```

Comment:
This method is called exactly 1 time and does not offer a lot.
It is likely better to drop the method and just write its contents here directly.

```ruby
storage_system = resource_named(:StorageSystem).new(@item.client, {credentials: { ip_hostname: @new_resource.storage_system }, name: @new_resource.storage_system})
@item.set_storage_system(storage_system)
```
Then use `storage_system['uri']` instead of `storage_system_uri` on the following calls.